### ISSUE TITLE
Buttons are not readable in high contrast mode (dark mode) on Firefox

### ISSUE THREAD

**Issue by Daniel-Schulz-Private** [carbon-design-system/carbon#4794]
Abstract: when switching Firefox to high contrast mode using a theme with black background, the labels of Carbon-Buttons become unreadable.

## Environment

> MacOS

> Firefox

> visual verification

## Detailed description

> problematic with both Carbon v9 as well as Carbon v10

> I expect the Button label to be readable

> I couldn't read the Button label

> Success Criterion 1.4.3 Contrast (Minimum)

## Steps to reproduce the issue

1. open Firefox
2. select "Preferences..."
3. In section "Language and Appearance" / "Fonts and Colors", select "Colors..."
4. For "Text", choose white as color
5. For "Background", select black color
6. For "Override the colors specified by the page with your selections above", select "Always"

7. Navigate to https://v9.carbondesignsystem.com/components/button/code or https://www.carbondesignsystem.com/components/button/code

You will see something like this (for v9):
![image](https://user-images.githubusercontent.com/51151716/69792190-71e12a00-11c6-11ea-91bf-ff620c6e8355.png)

> Please your your very own web page (URLs from above) as sandbox.## Additional information

**Comment by joshblack** [comment]
cc @dakahn if you have a chance to confirm

**Comment by abbeyhrt** [comment]
@joshblack this seems to be for v9, does that change anything?

**Comment by joshblack** [comment]
Hi @Daniel-Schulz-Private! 👋 Thanks so much for taking the time to make this issue 🙏 We do support WHCM in Chrome, Edge, and IE11, but we do not support this technique of setting HCM through browser preferences. Here's a [link](https://github.com/carbon-design-system/carbon/blob/master/docs/on-windows-hcm-support.md) for a full explainer of this policy. 

Let us know if you have any other questions! 😄
